File: pset6/read/readability.py
from cs50 import get_string

# get the text from the user
text = get_string("Text: ")

# set the variables for counting
L, W, S = 0, 1, 0

# set the list of special characters pointer
points = [".", "!", "?", ":"]

# iterate the string counting letters, words and sentences
for i in range(0, len(text), 1):
    if str.isalpha(text[i]):
        L += 1

    if str.isspace(text[i]) and str.isalpha(text[i + 1]):
        W += 1

    if text[i] in points:
        S += 1

# calculate de Coleman-Liau index
index = 5.88 * L / W - 29.6 * S / W - 15.8

# label the text based on the index
if index <= 0:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print("Grade ", round(index))

File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h> // for string lenght
#include <ctype.h> // for types
#include <math.h> // for round


int main(void)
{
    // Prompt for user's text
    string text = get_string("Text: ");

    //setting counters for LETTERS, WORDS and SENTENCES
    int L = 0;
    int W = 1;
    int S = 0;

    //Checking and counting the text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            L++;
        }

        if (isalpha(text[i + 1]) && isspace(text[i]))
        {
            W++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?' || text[i] == ':')
        {
            S++;
        }
    }

    //
    float lperw = 100 * (float) L / (float) W; //finding letters per 100 words
    float sperw = 100 * (float) S / (float) W; //finding sentences per 100 words
    float index = 0.0588 * (float) lperw - 0.296 * (float) sperw - 15.8; //finding Coleman-Liau index


    //printf("%i letters(s)\n", L);
    //printf("%i word(s)\n", W);
    //printf("%i sentences(s)\n", S);

    if (index <= 0)

    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)

    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }

}
